retriever/lib/repository.py

```diff
@@ -69,9 +73,14 @@ def check_for_updates():
                 try:
                     os.remove(os.path.normpath(os.path.join(HOME_DIR, "scripts", script_name)))
                     _download_from_repository("scripts/" + script_name,
-                                              os.path.normpath(os.path.join(SCRIPT_WRITE_PATH, script_name)))
+                                              os.path.normpath(os.path.join(SCRIPT_WRITE_PATH, script_name)),
+                                              repo)
                 except Exception as e:
                     print(e)
                     pass
     except:
         raise
+
+    if repo == RETRIEVER_REPOSITORY:
+        return
+    check_for_updates(RETRIEVER_REPOSITORY)
```

Comment:
When `check_for_updates()` is called, it would update from both the upstream repositories: recipes (for all the scripts), and retriever (for the five legacy scripts). The above logic simply calls the function twice, once for each repo.